pvlib/snowcoverage.py

```diff
@@ -0,0 +1,146 @@
+"""
+The ``snow`` module contains functions that model the effect of snow on
```

Comment:
Filename is still `snowcoverage.py`

Reply:
yea. I need to sort out the name, split up one test, move this into `losses.py`.

Reply:
In retrospect I'm not sure `losses.py` is a great idea. There are losses at many steps of the modeling process and `losses.py` is going to look like the junk drawer of loss models. So I'd be ok sticking with `snow.py` or `snowcoverage.py` in this PR.

Reply:
Hope about a losses package then?
```
*-+- pvlib
  |
  *-+- losses  # put API in __init__.py
    |
    *- snow_coverage.py
    |
    *- soiling.py
```

Reply:
In principle it might make sense to have soiling and snow as neighbors to IAM, whether that means including `iam.py` in Mark's `losses` package or something else.

Reply:
I don't think a subpackage solves the problem. We have spectral "corrections" in `atmosphere.py`, AOI "modifiers" in `iam.py`, DC "efficiency" in `pvsystem.py`, inverter "efficiency" in `pvsystem.py` (maybe soon `inverter.py`), temperature "corrections" `temperature.py`. I'm comfortable describing all of these as "losses" of some form.

Reply:
>  junk drawer of loss models.

Nice metaphor.

Editorial: Let me just state that I think the PVsyst-style "loss diagrams" do somewhat of a disservice, by applying the term "loss" to quantities of energy that couldn't have been generated in the first place. An example that's not a loss (IMO) is "temperature loss"; the physical reality is that cells are less efficient at higher temperature and hot air happens. There's only a "temperature loss" if the PV system could operate at lower temperature but doesn't.

If we follow the modeling process diagram, which is a guide not a rule, a module `dc_losses` could contain functions for wiring and mismatch losses, but not soiling, since soiling applies to irradiance not DC current. Debatable whether `snow_coverage` is an effect on DC current or on irradiance. 

It seems that pvlib will have multiple modules `soiling.py`, `snow.py`, `mismatch.py` or `dc_losses.py` etc. at some level, and the question is about grouping these modules into a `losses` folder, or not. @wholmgren's question suggests we should agree on a list of these modules, and maybe it will become more clear where to place them.

Here's an attempt to list categories for functions that could be broadly termed losses using my view of that term:

- irradiance reflection (iam.py)
- soiling
- snow
- shading (could overlap with DC losses)
- mismatch (cell to cell and module-to-module variation, also irradiance variation?)
- DC wiring
- inverter MPPT efficiency (not part of the current inverter models, but could be)
- tracker accuracy
- grid support functions (e.g., use of reactive power as voltage control)

Reply:
As an alternate to "losses", how about the term "effects" which can be either positive or negative deviations from the irradiance-corrected nominal condition?